core/domain/skill_services.py

```diff
@@ -232,156 +232,161 @@ def get_all_skill_summaries():
+    augmented_skill_summaries = []
+
+    augmented_skill_summaries_batch, urlsafe_start_cursor, more = (
+        _get_augmented_skill_summaries_in_batches(
+            num_items_to_fetch, cursor, sort_by))
+    filtered_augmented_skill_summaries = _filter_skills_by_status(
+        augmented_skill_summaries_batch, status)
+    filtered_augmented_skill_summaries = _filter_skills_by_classroom(
         filtered_augmented_skill_summaries, classroom_name)
-    filtered_augmented_skill_summaries = filter_skills_by_keywords(
+
+    filtered_augmented_skill_summaries = _filter_skills_by_keywords(
         filtered_augmented_skill_summaries, keywords)
-    skill_summaries.extend(filtered_augmented_skill_summaries)
+    augmented_skill_summaries.extend(filtered_augmented_skill_summaries)
 
-    while len(skill_summaries) < num_queries_to_fetch and more and next_cursor:
-        augmented_skill_summaries, next_cursor, more, total_count = (
-            get_all_skill_summaries_with_topic_and_classroom(
-                num_queries_to_fetch, next_cursor, sort_by))
+    while (len(augmented_skill_summaries) < num_items_to_fetch
+           and more and urlsafe_start_cursor):
+        augmented_skill_summaries_batch, urlsafe_start_cursor, more = (
+            _get_augmented_skill_summaries_in_batches(
+                num_items_to_fetch, urlsafe_start_cursor, sort_by))
 
-        filtered_augmented_skill_summaries = filter_skills_by_status(
-            augmented_skill_summaries, status)
-        filtered_augmented_skill_summaries = filter_skills_by_classroom(
+        filtered_augmented_skill_summaries = _filter_skills_by_status(
+            augmented_skill_summaries_batch, status)
+        filtered_augmented_skill_summaries = _filter_skills_by_classroom(
             filtered_augmented_skill_summaries, classroom_name)
-        filtered_augmented_skill_summaries = filter_skills_by_keywords(
+        filtered_augmented_skill_summaries = _filter_skills_by_keywords(
             filtered_augmented_skill_summaries, keywords)
-        skill_summaries.extend(filtered_augmented_skill_summaries)
+        augmented_skill_summaries.extend(filtered_augmented_skill_summaries)
 
-    return skill_summaries, next_cursor, more, total_count
+    return augmented_skill_summaries, urlsafe_start_cursor, more
 
 
-def get_all_skill_summaries_with_topic_and_classroom(
-        num_queries_to_fetch, cursor, sort_by):
+def _get_augmented_skill_summaries_in_batches(
+        num_items_to_fetch, cursor, sort_by):
     """Returns all the Augmented skill summaries after attaching
-       topic and classroom.
+    topic and classroom.
 
     Returns:
-        4-tuple(query_models, next_cursor, more, total_count). where:
-            skill_summaries: list(AugmentedSkillSummary). The list of
+        3-tuple(query_models, urlsafe_start_cursor, more). where:
+            augmented_skill_summaries: list(AugmentedSkillSummary). The list of
```

Comment:
augmented_skill_summaries is not used in your definition for the 3-tuple.

Reply:
Fixed.